src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -699,11 +703,6 @@ export default class Fms {
      * @param runwayModel {RunwayModel}
      */
     replaceDepartureProcedure(routeString, runwayModel) {
-        // this is the same procedure that is already set, no need to continue
-        if (this.hasLegWithRouteString(routeString)) {
```

Comment:
why not just move this `if` block below the runway check?  not doing this at all means we could, potentially, attempt to replace a given flightPlan with exactly the same thing.  this block gives us an early out in that case and saves some processing.